### ISSUE TITLE
A false negative about the rule HE_HASHCODE_NO_EQUALS and an official testcase

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1765]
**SpotBugs Version:** 4.4.2
**Rule:** [HE_HASHCODE_NO_EQUALS](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#he-class-defines-hashcode-but-not-equals-he-hashcode-no-equals)

Hi! I find a false negative about this rule. SpotBugs should have reported a warning at line 3. Besides, I think this case is a little simple, so I try to get some test cases in official repo and find  [this case](https://github.com/spotbugs/spotbugs/blob/master/spotbugsTestCases/src/java/hashCODEnoEQUALS.java) is also cannot be detected. We may need more testing for this rule.
```java
public class clazz {
  @Override
  public int hashCode() {
      return 476578;
  }
}
```
**Expected outcome:**
SpotBugs should report a violation at line 4, but doesn't. This is a false-negative.

**Running SpotBugs through:** [CLI]
